File: frontend/src/types/reports.ts
export interface EmployeeHours {
  employee_id: number
  first_name: string
  last_name: string
  total_hours: number
}

export interface HoursReport {
  employees: EmployeeHours[]
  total_hours: number
}

export interface WeeklyReport {
  week_start: string
  total_hours: number
  employees: EmployeeHours[]
}

export interface MonthlyReport {
  month: string
  total_hours: number
  employees: EmployeeHours[]
}

File: backend/app/schemas/reports.py
from pydantic import BaseModel
from typing import List


class EmployeeHours(BaseModel):
    employee_id: int
    first_name: str
    last_name: str
    total_hours: float


class WeeklyReport(BaseModel):
    week_start: str
    total_hours: float
    employees: List[EmployeeHours]


class MonthlyReport(BaseModel):
    month: str
    total_hours: float
    employees: List[EmployeeHours]


class HoursReport(BaseModel):
    employees: List[EmployeeHours]
    total_hours: float
